.gitlab/functional_test/regression_detector.yml

```diff
@@ -208,3 +208,207 @@ single-machine-performance-regression_detector-pr-comment:
         exit 0
       fi
       exit $exitcode
+
+single-machine-performance-quality-gates-only:
+  stage: functional_test
+  timeout: 1h10m
+  rules:
+    - !reference [.except_main_or_release_branch]
+    - when: on_success
+  image: registry.ddbuild.io/ci/datadog-agent-buildimages/docker_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  needs:
+    - job: single_machine_performance-amd64-a7
+      artifacts: false
+  artifacts:
+    expire_in: 1 weeks
+    paths:
+      - submission_metadata # for provenance, debugging
+      - ${CI_COMMIT_SHA}-baseline_sha # for provenance, debugging
+      - outputs/report.md # for debugging, also on S3
+      - outputs/regression_signal.json # for debugging, also on S3
+      - outputs/bounds_check_signal.json # for debugging, also on S3
+      - outputs/junit.xml # for debugging, also on S3
+    when: always
+  variables:
+    SMP_VERSION: 0.18.1
+  # Single Machine Performance Quality Gates are mandatory.
+  allow_failure: false
+  script:
+    # Ensure output files exist for artifact downloads step
```

Comment:
nit: you can save the duplicated parts on an `extends` part or a `[!reference]`

Reply:
Agreed. I think refactoring the configuration using `extends` would be premature. The copy-paste implementation was intended to be a stopgap while #31017 is being finalized so that we could evaluate what would happen if we made quality gates jobs mandatory. If this PR is merged before #31017, then #31017 would be modified to delete these quality gates jobs in the Agent CI configuration.